src/services/completions.ts

```diff
@@ -2831,6 +2849,27 @@ namespace ts.Completions {
 
         if (!contextToken) return undefined;
 
+        function isPropertyBeforeConstructorInitalized(classDecl: ClassDeclaration, _constructor: ClassElement) {
+            const indexOfConstructor = classDecl.members.indexOf(_constructor);
+            if (indexOfConstructor <= 0) {
+                return true;
+            }
+            return isInitializedProperty(classDecl.members[indexOfConstructor - 1]);
+        }
+
+        // class c { prop \n constructor| }
+        // class c { prop: number \n constructor| }
+        // class c { prop = somewhat \n constructor| }
+        // and also for `;` replacing `\n`
+        if (location.kind === SyntaxKind.ConstructorKeyword && (

```

Comment:
It looks like this whole condition applies only once you’ve typed the entire `constructor` keyword, and stops applying as soon as you’ve typed the open parenthesis afterwards, which is not really a time when completions are likely to be triggered, or when they will be valuable. This might be technically correct, but I’m confused about what purpose it’s serving.

Reply:
This is the fix for the most original issue. https://github.com/microsoft/TypeScript/issues/43650
Manually triggered completion (Ctrl+Space in Windows) doesn't offer `constructor` for this:
```ts
class C {
  blah // w/ w/o semicolon, type, initializer
  constructor/**/ // must be fully present
}
```
The cause seems to be a missing case in `tryGetObjectTypeDeclarationCompletionContainer`.

Reply:
I think I disagree with the premise that #43650 is worth more than a line of code to fix. From the original issue opened against VS Code, it looks like someone was just hunting for inconsistencies, not experiencing a legitimate UX issue. I’m not against fixing it, but I am against fixing it if no one can explain why it matters and it takes a multiline `if` statement, a bunch of explanatory comments, and a dozen fourslash tests.

Reply:
:laughing: Yep. The latest fix is of 4 LOC.